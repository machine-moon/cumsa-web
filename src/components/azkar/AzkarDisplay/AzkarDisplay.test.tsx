import React from "react";
import { render } from "@testing-library/react";

type Zikr = {
  ID: number;
  ARABIC_TEXT: string;
  LANGUAGE_ARABIC_TRANSLATED_TEXT: string;
  TRANSLATED_TEXT: string;
  REPEAT: number;
  AUDIO: string;
};

const MockAzkarDisplay = ({ azkar, loading }: { azkar: Zikr[]; loading: boolean }) => {
  if (loading) {
    return (
      <div data-testid="azkar-display-loading">
        <div>Loading...</div>
      </div>
    );
  }

  if (!azkar.length) {
    return (
      <div data-testid="azkar-display-empty">
        <div>Introduction text</div>
      </div>
    );
  }

  return (
    <div data-testid="azkar-display">
      <div data-testid="language-toggle">Language Toggle</div>
      {azkar.map((zikr) => (
        <div key={zikr.ID} data-testid="zikr-item">
          {zikr.TRANSLATED_TEXT}
        </div>
      ))}
    </div>
  );
};

describe("AzkarDisplay", () => {
  const mockAzkar = [
    {
      ID: 1,
      ARABIC_TEXT: "سُبْحَانَ اللهِ",
      LANGUAGE_ARABIC_TRANSLATED_TEXT: "Subhanallah",
      TRANSLATED_TEXT: "Glory is to Allah",
      REPEAT: 33,
      AUDIO: "",
    },
    {
      ID: 2,
      ARABIC_TEXT: "الْحَمْدُ للهِ",
      LANGUAGE_ARABIC_TRANSLATED_TEXT: "Alhamdulillah",
      TRANSLATED_TEXT: "Praise is to Allah",
      REPEAT: 33,
      AUDIO: "",
    },
  ];

  it("renders correctly with azkar", () => {
    const { container } = render(<MockAzkarDisplay azkar={mockAzkar} loading={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly when loading", () => {
    const { container } = render(<MockAzkarDisplay azkar={[]} loading={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with empty azkar (introduction)", () => {
    const { container } = render(<MockAzkarDisplay azkar={[]} loading={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
