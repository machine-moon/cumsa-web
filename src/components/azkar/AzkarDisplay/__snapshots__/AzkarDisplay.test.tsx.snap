// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzkarDisplay renders correctly when loading 1`] = `
<div
  data-testid="azkar-display-loading"
>
  <div>
    Loading...
  </div>
</div>
`;

exports[`AzkarDisplay renders correctly with azkar 1`] = `
<div
  data-testid="azkar-display"
>
  <div
    data-testid="language-toggle"
  >
    Language Toggle
  </div>
  <div
    data-testid="zikr-item"
  >
    Glory is to Allah
  </div>
  <div
    data-testid="zikr-item"
  >
    Praise is to Allah
  </div>
</div>
`;

exports[`AzkarDisplay renders correctly with empty azkar (introduction) 1`] = `
<div
  data-testid="azkar-display-empty"
>
  <div>
    Introduction text
  </div>
</div>
`;
